Packs/ApiModules/Scripts/SiemApiModule/SiemAPIModule.py

```diff
@@ -0,0 +1,161 @@
+# pylint: disable=no-name-in-module
+# pylint: disable=no-self-argument
+
+from abc import ABC
+from typing import Any, Optional
+import demistomock as demisto
+from CommonServerPython import *
+from CommonServerUserPython import *
+
+from enum import Enum
+from pydantic import BaseConfig, BaseModel, AnyUrl, validator  # type: ignore[E0611, E0611, E0611]
+from requests.auth import HTTPBasicAuth
+
+
+class Method(str, Enum):
+    GET = 'GET'
+    POST = 'POST'
+    PUT = 'PUT'
+    HEAD = 'HEAD'
+    PATCH = 'PATCH'
+    DELETE = 'DELETE'
+
+
+def load_json(v: Any) -> dict:
+    if not isinstance(v, (dict, str)):
+        raise ValueError('headers are not dict or a valid json')
+    if isinstance(v, str):
+        try:
+            v = json.loads(v)
+            if not isinstance(v, dict):
+                raise ValueError('headers are not from dict type')
+        except json.decoder.JSONDecodeError as exc:
+            raise ValueError('headers are not valid Json object') from exc
+    if isinstance(v, dict):
+        return v
+
+
+class IntegrationHTTPRequest(BaseModel):
+    method: Method
+    url: AnyUrl
+    verify: bool = True
+    headers: dict = dict()  # type: ignore[type-arg]
+    auth: Optional[HTTPBasicAuth]
+    data: Any = None
+
+    class Config(BaseConfig):
+        arbitrary_types_allowed = True
+
+    _normalize_headers = validator('headers', pre=True, allow_reuse=True)(
+        load_json
+    )
+
+
+class Credentials(BaseModel):
+    password: str
+    identifier: Optional[str]
```

Comment:
readability:
username-password combo vs password-username - second sound weird.
like PB and J, not J and PB.
like Captain and Tennille, not Tennille and Captain
```suggestion
    identifier: Optional[str]
    password: str
```